qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/datamodel/TemplateDataSupport.java

```diff
@@ -0,0 +1,161 @@
+ * 			return price.multiply(new BigDecimal("0.9"));
+ * 		}
+ * }
+ * </code>
+ *
+ *
+ * @see https://quarkus.io/guides/qute-reference#template_data
+ *
+ */
+public class TemplateDataSupport extends AbstractAnnotationTypeReferenceDataModelProvider {
+
+	private static final Logger LOGGER = Logger.getLogger(TemplateDataSupport.class.getName());
+
+	private static final String[] ANNOTATION_NAMES = {
+			TEMPLATE_DATA_ANNOTATION
+
+	};
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	@Override
+	protected void processAnnotation(IJavaElement javaElement, IAnnotation annotation, String annotationName,
+			SearchContext context, IProgressMonitor monitor) throws JavaModelException {
+		if (!(javaElement instanceof IAnnotatable)) {
+			return;
+		}
+		IAnnotation templateData = AnnotationUtils.getAnnotation((IAnnotatable) javaElement, TEMPLATE_DATA_ANNOTATION);
+		if (templateData == null) {
+			return;
+		}
+		if (javaElement instanceof IType) {
+			IType type = (IType) javaElement;
+			collectResolversForTemplateData(type, templateData, context.getDataModelProject().getValueResolvers(),
+					monitor);
+		}
+	}
+
+	private static void collectResolversForTemplateData(IType type, IAnnotation templateData,
+			List<ValueResolverInfo> resolvers, IProgressMonitor monitor) {
+		try {
+			ITypeResolver typeResolver = QuteSupportForTemplate.createTypeResolver(type);
+			IField[] fields = type.getFields();
+			for (IField field : fields) {
+				// May want to check method access? i.e. only resolve public
+				IAnnotation fieldTemplateData = AnnotationUtils.getAnnotation(field, TEMPLATE_DATA_ANNOTATION);
+				collectResolversForTemplateData(field, fieldTemplateData != null ? fieldTemplateData : templateData,
+						resolvers, typeResolver);
+			}
+			IMethod[] methods = type.getMethods();
+			for (IMethod method : methods) {
+				// May want to check method access? i.e. only resolve public
+				IAnnotation methodTemplateData = AnnotationUtils.getAnnotation(method, TEMPLATE_DATA_ANNOTATION);
+				collectResolversForTemplateData(method, methodTemplateData != null ? methodTemplateData : templateData,
+						resolvers, typeResolver);
+			}
+		} catch (JavaModelException e) {
+			LOGGER.log(Level.SEVERE, "Error while getting methods of '" + type.getElementName() + "'.", e);
+		}
+	}
+
+	private static void collectResolversForTemplateData(IField field, IAnnotation templateData,
+			List<ValueResolverInfo> resolvers, ITypeResolver typeResolver) {
+		ValueResolverInfo resolver = new ValueResolverInfo();
+		resolver.setSourceType(field.getDeclaringType().getFullyQualifiedName());
+		resolver.setSignature(typeResolver.resolveFieldSignature(field));
+		try {
+			String namespace = AnnotationUtils.getAnnotationMemberValue(templateData,
+					TEMPLATE_EXTENSION_ANNOTATION_NAMESPACE);
+			String matchName = AnnotationUtils.getAnnotationMemberValue(templateData,
+					TEMPLATE_EXTENSION_ANNOTATION_MATCH_NAME);

```

Comment:
This code is valid for @TemplateExtenion but not for @TemplateData, please remove matchName